pkg/cmd/auth/shared/login_flow_test.go

```diff
@@ -282,6 +289,61 @@ func TestLogin(t *testing.T) {
 	}
 }
 
+func TestAuthenticatingGitCredentials(t *testing.T) {
+	// Given we have no host or global credential helpers configured
+	// And given they have chosen https as their git protocol
+	// When they choose to authenticate git with their GitHub credentials
+	// Then gh is configured as their credential helper for that host
+	ios, _, _, _ := iostreams.Test()
+
+	reg := &httpmock.Registry{}
```

Comment:
I'm happy to look into figuring out how this test would fit into the table tests but I wanted it to be **very** clear in exactly what it was doing for this review.